assignments/sidharth/week1/1.7.txt

```diff
@@ -0,0 +1,13 @@
+The problem with finding square root of very small numbers using this method is that The logic prevents the improvement of the guess past the 0.001 hardcoded threshold since the radicant itself is much smaller than this threshold.
+
+The problem with finding square root of very large numbers is that the code never stops running. This is because floats have a fixed precision. After a certain point, the guesses are unable to guess the square root of a number within 0.001 of the answer. At that point, they alternate between the two guesses, both of whose squares are more than 0.001 away from the argument.
```

Comment:
nit: as discussed on the mailing list, another issue is when (improve) falls under floating point epsilon, resulting in a noop causing an infinite recursion.